api/app/services/event_service.rb

```diff
@@ -16,10 +16,12 @@ def self.fetch(params)
     self.paginate(events, params)
   end
 
-  def self.create(params)
+  def self.create(params, user)
```

Comment:
Using bang method create! in service layer will raise on failure, potentially causing application crash. Rescue exceptions and return meaningful errors.